pipenv/core.py

```diff
@@ -658,47 +656,49 @@ def _cleanup_procs(project, procs, failed_deps_queue, retry=True):
             click.secho(out.strip() or err.strip(), fg="cyan")
         # The Installation failed...
         if failed:
-            # If there is a mismatch in installed locations or the install fails
-            # due to wrongful disabling of pep517, we should allow for
-            # additional passes at installation
-            if "does not match installed location" in err:
-                project.environment.expand_egg_links()
-                click.echo(
-                    "{}".format(
-                        click.style(
-                            "Failed initial installation: Failed to overwrite existing "
-                            "package, likely due to path aliasing. Expanding and trying "
-                            "again!",
-                            fg="yellow",
+            deps = c.deps.copy() if hasattr(c, "deps") else None
+            for dep in deps:
```

Comment:
I think if `deps is None`, this loop will cause error.

Reply:
Yes, that's correct. If deps is `None` there will be an error. However, it's always setup to an empty list `[]`.

Reply:
Do you mean `deps = getattr(c, "deps", []).copy()` ?

Reply:
A `[]` won't have a copy method,  so it will break.